k8s/configutils/configutil.go

```diff
@@ -119,13 +119,19 @@ func ValidateObjectName(name string, rootResource string) *field.Error {
 		return field.Invalid(field.NewPath("spec").Child("rootResource"), rootResource, "rootResource must be a non-empty string")
 	}
 
-	parts := strings.Split(name, "-")
-	if len(parts) != 2 {
-		return field.Invalid(field.NewPath("name"), name, "name must be in the format of <containerName>-<version> and only one hyphen is allowed")
+	if !strings.HasPrefix(name, rootResource) {
+		return field.Invalid(field.NewPath("name"), name, "name must start with spec.rootResource")
 	}
 
-	if parts[0] != rootResource {
-		return field.Invalid(field.NewPath("name"), name, "name must start with spec.rootResource")
+	prefix := rootResource + "-v-"
+	remaining := strings.TrimPrefix(name, prefix)
+
+	if len(remaining) == 0 {
+		return field.Invalid(field.NewPath("name"), name, "name should be in the format '<rootResource>-v-<version>'")
+	}
+
+	if strings.Contains(remaining, "-v-") || strings.HasPrefix(remaining, "v-") {
```

Comment:
Not sure if this check is necessary? container1-v-v-1 should be okay for a "v-1" object under "container1"?

Reply:
It will conflict with container "container1-v" and version `1` resource from ARM.

Reply:
Not exactly for now. Seperator to resource name is straight-forward, just replace ":" to what defined for separator. But not for resource name to reference style:
One case is `handleSolution` in `instance_controller`. Watch all solution changes and match name to `instance.spec.solution`. If we do not have any constraints, then we should change `watcher` logic to something like: store  `version` in webhook: `spec.version` = `name` trim `rootResource` (required property). We have everything in versioned objects. Then we can combine new referred solutionName as `rootResource`:"version". And using this to matching list option. Should be good?

Reply:
We still care about conflicting naming with `<roortResource>+<seperator>+<version>`. So "-" maybe is not enough to release the constraint of blocking `-` in all resources. So `-v-` could be a better choice.